### ISSUE TITLE
rustdoc does not warn when linking to an item with no generated docs

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#81979]
I tried this code:

```rust
//! Link to [S]

#[doc(no_inline)]
pub use inner::S;

mod inner {
    pub struct S;
}
```

I expected to see this happen: Rustdoc warns that the links will all be broken

Instead, this happened: Rustdoc gives no warning and generates HTML that looks bad: 
![image](https://user-images.githubusercontent.com/23638587/107592146-361f5e00-6bda-11eb-9a58-4670becc2e79.png)

### Meta

`rustdoc --version`: rustdoc 1.52.0-nightly (9778068cb 2021-02-07)
